compiler/rustc_attr_parsing/src/parser.rs

```diff
@@ -495,12 +495,9 @@ impl<'a> MetaItemListParserContext<'a> {
                 // where the macro didn't expand to a literal. An error is already given
                 // for this at this point, and then we do continue. This makes this path
                 // reachable...
-                let e = self.dcx.span_delayed_bug(
-                    *span,
-                    "expr in place where literal is expected (builtin attr parsing)",
-                );
-
-                return Some(MetaItemOrLitParser::Err(*span, e));
+                // NOTE: For backward compatibility (https://github.com/rust-lang/rust/issues/137687#issuecomment-2816312274),
```

Comment:
Nit: probably instead use the split-off https://github.com/rust-lang/rust/issues/140612 now, since #137687 is tracking the original different ICE.